osu.Game/Gamemode/Gamemode.cs

```diff
@@ -0,0 +1,34 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using System.Runtime.InteropServices;
+
+namespace Gamemode
+{
+    public static class GamemodeRequest {
```

Comment:
Platform-specific stuff like this shouldn't reside in `osu.Game` project but rather in `osu.Desktop` project  and use a component that you insert into the drawable hierachy and which listens for value changes of `Game.LocalUserPlaying` field instead of calling it in `PlayerLoader` (see the following link for an example https://github.com/ppy/osu/blob/master/osu.Desktop/Windows/GameplayWinKeyBlocker.cs).

Also you may want to add platform checks to add this component only when the host runs linux and should add handling for cases where the DLL isn't locatable (as it'll cause a crash as it is right now when calling the PInvoke methods and the DLL isn't locatable).

Reply:
Will look into moving it to the osu.Desktop project.

It already checks whether the DLL is available on the system. Tried to run without gamemode on and it works without any issues.